### PR TITLE
Reordered file structure to better seperate the backend into layers

### PR DESCRIPTION
**Pull request by NikhilRaikar17**

### CODE REVIEW

File: backend/requirements.txt [hunk 1]
```diff
@@ -1,10 +1,132 @@
-langchain_core
```
Comment (tmeftah) [inline backend/requirements.txt]: please add only this packages in the requirements (i tested them):
langchain
langchain_community
langchain_core
langchain_chroma
langchain_ollama
langfuse
sqlalchemy
fastapi[standard]
pyjwt
python-dotenv

**Review by tmeftah — CHANGES_REQUESTED** [review]
please add only this packages in the requirements (i tested them):
langchain
langchain_community
langchain_core
langchain_chroma
langchain_ollama
langfuse
sqlalchemy
fastapi[standard]
pyjwt
python-dotenv

File: backend/service/query_service.py [hunk 2]
```diff
@@ -1 +1,27 @@
...
\ No newline at end of file
+import json
+import uuid
+from fastapi.responses import StreamingResponse
+
+from backend.rag_llms_langchain import chain
+from backend.embeddings.ingest import get_vectorstore
+
+
+def query_service(query: str):
```
Comment (tmeftah) [inline backend/service/query_service.py:9]: add async

**Review by tmeftah — CHANGES_REQUESTED** [review]
add async  befor function

File: backend/service/query_service.py (hunk 2 above)
Comment (NikhilRaikar17) [reply backend/service/query_service.py]: Added async

File: backend/requirements.txt (hunk 1 above)
Comment (NikhilRaikar17) [reply backend/requirements.txt]: Updated req file

**Pull request closed without merging**